_docs/setup/cluster-extension.md

```diff
@@ -0,0 +1,130 @@
+---
+title: Istio Kubernetes Cluster Extension
+overview: Instructions to add external machines to your Istio managed Kubernetes cluster.
+
+order: 25
+
+layout: docs
+type: markdown
+---
+
+This task shows how to install Istio in a non-kubernetes machine in the same network
+(VPC, VPN) as the Kubernetes cluster.
+
+_This document is under construction._
+
+## Prerequisites
+
+* You have [installed Istio](install-kubernetes.html) on your cluster
+
+* The machine must have IP connectivity to the endpoints in the cluster. This
+typically requires same VPC or a VPN connection, as well as a container network that
+provides direct (without NAT or firewall deny) routing to the endpoints. The machine
+is not required to have access to the cluster IP addresses assigned by K8S.
+Istio's Pilot and the Istio sidecar and configuration installed on the external
+machine(s) by following these steps provide the service level connectivity and
+all the Istio advance routing and control capabilities.
+
+* The control plane (Pilot, Mixer, CA) and Kubernetes DNS server must be accessible
+from the VM. This is typically done using an [Internal Load
+Balancer](https://kubernetes.io/docs/concepts/services-networking/service/#internal-load-balancer).
+
+## Installation steps
+
+Setup consists of preparing the cluster for extensions and installing and configuring each VM.
+
+Example functions to help with the cluster setup are available in
+[istio_vm_common.sh](https://raw.githubusercontent.com/istio/istio/master/install/tools/istio_vm_common.sh).
+Depending on cluster and environment you may need to adjust them for your cluster and VM.
+The scripts need to be run by a cluster admin, with permissions to Kubernetes and the VM.
+
+An example script to configure a machine is available in [istio_vm_setup.sh](https://raw.githubusercontent.com/istio/istio/master/install/tools/istio_vm_setup.sh).
+You should customize it to match your normal provisioning tools and DNS requirements.
+
+### Preparing the K8s cluster
+
+* Setup internal load balancers for Kube DNS, Pilot, Mixer and CA. This step is specific to
+each cluster - the istioInitILB function has an example for GKE using internal load balancers.
```

Comment:
missing context.. what is istioInitILB ? even if you add a link to the script, it does not explain the setup.